test/jasmine/tests/transition_test.js

```diff
@@ -106,7 +106,7 @@ describe('Plots.transition', function() {
             var pythonLogo = 'https://images.plot.ly/language-icons/api-home/python-logo.png';
 
             function imageel() {
-                return gd._fullLayout._imageUpperLayer.select('image').node();
```

Comment:
I wasn't able to replicate the transition error on real code so started poking around the test itself. I don't understand it, but somehow doing this d3 selection was resetting the data stored with the image element, causing the key function I added in this PR to think the existing element wasn't the right one to use for the new image, so line 135 below `expect(e1).toBe(e2);` failed. Switching to a native JS `querySelector` fixed it. 🤷